test/reducedim.jl

```diff
@@ -50,14 +50,14 @@ safe_minabs(A::Array{T}, region) where {T} = safe_mapslices(minimum, abs.(A), re
     fill!(r, -1.5)
     @test minimum!(abs, r, Areduc, init=false) ≈ fill!(r2, -1.5)
 
-    @test sum(Areduc, dims=region) ≈ safe_sum(Areduc, region)
-    @test prod(Areduc, dims=region) ≈ safe_prod(Areduc, region)
-    @test maximum(Areduc, dims=region) ≈ safe_maximum(Areduc, region)
-    @test minimum(Areduc, dims=region) ≈ safe_minimum(Areduc, region)
-    @test sum(abs, Areduc, dims=region) ≈ safe_sumabs(Areduc, region)
-    @test sum(abs2, Areduc, dims=region) ≈ safe_sumabs2(Areduc, region)
-    @test maximum(abs, Areduc, dims=region) ≈ safe_maxabs(Areduc, region)
-    @test minimum(abs, Areduc, dims=region) ≈ safe_minabs(Areduc, region)
+    @test @inferred sum(Areduc, dims=region) ≈ safe_sum(Areduc, region)
+    @test @inferred prod(Areduc, dims=region) ≈ safe_prod(Areduc, region)
```

Comment:
Does this `@inferred` apply to the approx operator or the LHS?

Reply:
I guess it applies to the `isapprox`:
```julia
julia> f(x) = x < 0 ? '1' : 1
f (generic function with 1 method)

julia> using Test

julia> @inferred f(1)
ERROR: return type Int64 does not match inferred return type Union{Char, Int64}
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at none:0

julia> @inferred f(1) == 1
true

julia> @inferred(f(1)) == 1
ERROR: return type Int64 does not match inferred return type Union{Char, Int64}
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at none:0
```